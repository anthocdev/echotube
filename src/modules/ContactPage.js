import React from "react";
import "../style/contact.css";
import axios from "axios";
/* Rules page place holder - Currently one of the routes for testing */
class ContactPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      email: "",
      message: "",
      emailError: "",
      nameError: "",
      messageError: "",
    };
  }

  //Validation for fields
  validateData = () => {
    let emailError = "";
    let nameError = "";
    let messageError = "";

    let valid = true;
    if (!this.state.email.includes("@")) {
      emailError = "Invalid email address";
      valid = false;
    }

    if (!this.state.email) {
      emailError = "Can't be blank";
      valid = false;
    }

    if (this.state.message.length < 100) {
      messageError = "Must contain at least 100 characters.";
      valid = false;
    }

    if (!this.state.message) {
      messageError = "Can't be blank";
      valid = false;
    }

    if (!this.state.name) {
      nameError = "Can't be blank";
      valid = false;
    }

    this.setState({ emailError });
    this.setState({ messageError });
    this.setState({ nameError });

    return valid;
  };

  //Overriding submission event
  handleSubmit = (event) => {
    console.log(event);
    event.preventDefault();

    const validData = this.validateData();

    if (validData) {
      let dataToSend = {
        name: this.state.name,
        email: this.state.email,
        message: this.state.message,
      };
      this.sendToServer(dataToSend);
    }
  };

  async sendToServer(sendData) {
    await axios({
      method: "POST",
      url: "http://localhost:3001/api/sendmail",
      data: sendData,
    }).then((response) => {
      if (response.status === 200) {
        alert("Your inquiry has been received.");
        this.clearForm();
      } else if (response.status === 500) {
        alert("Your inquiry has failed to arrive.");
      }
    });
  }

  clearForm() {
    this.setState({ name: "", email: "", message: "" });
  }

  onChange(event) {
    console.log(event);
    this.setState({ [event.target.name]: event.target.value });
  }

  render() {
    return (
      <div className="contact">
        <a>Contact Page Placeholder</a>
        <form
          id="contactForm"
          onSubmit={this.handleSubmit.bind(this)}
          method="POST"
        >
          <div className="formItem">
            <label htmlFor="name">Name</label>
            <input
              type="text"
              name="name"
              className="form-control"
              onChange={this.onChange.bind(this)}
              value={this.state.name}
            />
            <div className="errorMessage">{this.state.nameError}</div>
          </div>
          <div className="formItem">
            <label htmlFor="exampleInputEmail1">Email address</label>
            <input
              type="email"
              name="email"
              className="formItem"
              aria-describedby="emailHelp"
              onChange={this.onChange.bind(this)}
              value={this.state.email}
            />
            <div className="errorMessage">{this.state.emailError}</div>
          </div>
          <div className="formItem">
            <label htmlFor="message">Message</label>
            <textarea
              className="form-control"
              rows="6"
              name="message"
              onChange={this.onChange.bind(this)}
              value={this.state.message}
            ></textarea>
            <div className="errorMessage">{this.state.messageError}</div>
          </div>
          <button type="submit" className="btn btn-primary">
            Submit
          </button>
        </form>
      </div>
    );
  }
}

export default ContactPage;
