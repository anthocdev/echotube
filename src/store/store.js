import { applyMiddleware, createStore, compose } from "redux";
import { createLogger } from "redux-logger";
import { default as thunk } from "redux-thunk";
import { createPromise } from "redux-promise-middleware";
import axios from "axios";

import reducer from "../reducers";
//Pulling JWT Token from local storage
const jwttoken = localStorage.getItem("JWT_TOKEN");
//Setting JWT Token as common header for axios
axios.defaults.headers.common["Authorization"] = jwttoken;
//Create store with middleware included
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; //Adding redux dev tool compatibility
const middleware = applyMiddleware(createPromise(), thunk, createLogger()); //Adding thunk and console logging for testing as middleware

export default createStore(
  reducer,
  {
    //Initializing auth state if JWT token is available
    auth: {
      jwtToken: jwttoken,
      isAuth: jwttoken ? true : false
    }
  },
  composeEnhancers(middleware)
);
