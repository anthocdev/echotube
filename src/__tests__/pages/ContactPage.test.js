import React from "react";
import ContactPage from "../../modules/ContactPage";
import { Provider } from "react-redux";
import store from "../../store/store";
import { create, act } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Contact Page Tests", () => {
  let root;

  it("Renders Without Crashing", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <ContactPage />
          </Provider>
        </BrowserRouter>
      );
    });
  });

  test("Renders Matches Snapshot", async () => {
    act(() => {
      root = create(
        <BrowserRouter>
          <Provider store={store}>
            <ContactPage />
          </Provider>
        </BrowserRouter>
      );
    });
    expect(root).toMatchSnapshot();
  });
});
