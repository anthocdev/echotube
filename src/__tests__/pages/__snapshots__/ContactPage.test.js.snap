// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page Tests Renders Matches Snapshot 1`] = `
<div
  className="contact"
>
  <form
    className="contactForm"
    id="contactForm"
    method="POST"
    onSubmit={[Function]}
  >
    <div
      className="formItem"
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
      <div
        className="errorMessage"
      >
        
      </div>
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="exampleInputEmail1"
      >
        Email address: 
      </label>
       
      <br />
      <input
        aria-describedby="emailHelp"
        className="formItem"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
      <br />
      <div
        className="errorMessage"
      >
        
      </div>
    </div>
    <div
      className="formItem"
    >
      <label
        htmlFor="message"
      >
        Message
      </label>
      <textarea
        className="form-control"
        name="message"
        onChange={[Function]}
        rows="6"
        value=""
      />
      <div
        className="errorMessage"
      >
        
      </div>
    </div>
    <button
      className="btn btn-primary"
      style={
        Object {
          "backgroundColor": "purple",
          "borderColor": "purple",
          "marginTop": "20px",
        }
      }
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
